lib/services/cloud-service.ts

```diff
@@ -97,4 +116,30 @@ export abstract class CloudService extends EventEmitter {
 
 		return targetFileNames;
 	}
+
+	protected async getCollectedLogs(serverResponse: IServerResponse): Promise<string> {
+		try {
+			return this.$nsCloudOutputFilter.filter(await this.$nsCloudS3Service.getContentOfS3File(serverResponse.outputUrl));
+		} catch (err) {
+			this.$logger.warn(`Unable to get cloud operation output. Error is: ${err}`);
+		}
+	}
+
+	protected getResult(cloudOperationId: string): ICloudOperationResult {
+		try {
+			return this.cloudOperations[cloudOperationId].getResult();
+		} catch (err) {
+			this.$logger.trace(err);
```

Comment:
Shouldn't we throw the error?

Reply:
We use this method only if there is error. If there is no result (the result file is empty or we can't get it), we want to throw the original error, not the getResult error. If we throw the error here, we need to trace the getResult error in every cloud service which uses this method. Do you think it will be better to throw the error here?